### PR TITLE
[uuid] improve perf

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Follows https://dev.to/fredbouchery/performance-des-uuids-29dj

### CODE REVIEW

File: src/Uuid/Uuid.php [hunk 1]
```diff
@@ -146,7 +146,7 @@ public static function uuid_is_valid($uuid)
             return null;
         }
 
-        return null !== self::uuid_parse_as_array($uuid);
+        return (bool) self::isValid($uuid);
```
Comment (lyrixx) [inline src/Uuid/Uuid.php]: let's cast in bool in `isValid` instead. It feels more natural to me that `isValid` returns a bool

File: src/Uuid/Uuid.php [hunk 2]
```diff
@@ -305,40 +303,31 @@ public static function uuid_unparse($uuidAsBinary)
...
         }
 
-        $uuid = sprintf('%s-%s-%s-%s-%s',
-            substr($data, 0, 8),
-            substr($data, 8, 4),
-            substr($data, 12, 4),
-            substr($data, 16, 4),
-            substr($data, 20, 12)
-        );
+        $uuidAsBinary = bin2hex($uuidAsBinary);
```
Comment (stof) [inline src/Uuid/Uuid.php]: Overriding the variable `$uuidAsBinary` to contain the uuid as hex looks wrong to me.

File: src/Uuid/Uuid.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Uuid/Uuid.php]: updated

File: src/Uuid/Uuid.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Uuid/Uuid.php]: updated

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**